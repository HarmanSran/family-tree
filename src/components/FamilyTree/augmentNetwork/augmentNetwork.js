const augmentEdges = (network) => {
  const { edges } = network;
  return edges.map((edge) => ({
    ...edge,
    label: undefined,
    color: ['husband', 'wife'].includes(edge.label) ? 'rgba(255, 117, 61)' : undefined,
  }));
};

const getNodeLabel = ({ name, metadata: { birth, death } }) => {
  const birthYear = birth ? birth.split('-')[0] : null;
  const deathYear = death ? death.split('-')[0] : null;
  let lifespan = null;
  if (birthYear && deathYear) {
    lifespan = `${birthYear} - ${deathYear}`;
  } else if (birthYear) {
    lifespan = `${birthYear} -`;
  } else if (deathYear) {
    lifespan = `- ${deathYear}`;
  }
  return `${name}${lifespan ? `\n${lifespan}` : ''}`;
};

const getNodeColor = ({ metadata: { gender } }) => (gender === 'F' ? 'rgba(255, 189, 248)' : 'rgba(133, 222, 255)');

const augmentNodes = (network) => {
  const { nodes } = network;
  return nodes.map(({ id, level, details }) => ({
    id,
    level,
    label: getNodeLabel(details),
    color: getNodeColor(details),
  }));
};

const augmentNetwork = (network) => ({
  nodes: augmentNodes(network),
  edges: augmentEdges(network),
});

export default augmentNetwork;
