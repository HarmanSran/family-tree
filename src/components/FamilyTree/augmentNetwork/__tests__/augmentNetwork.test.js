import augmentNetwork from '../augmentNetwork';
import network from './network';

describe('augmentNetwork', () => {
  it('augments nodes and edges as expected', () => {
    expect(augmentNetwork(network)).toMatchSnapshot();
  });
});
