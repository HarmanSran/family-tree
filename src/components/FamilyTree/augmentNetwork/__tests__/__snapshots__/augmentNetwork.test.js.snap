// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`augmentNetwork augments nodes and edges as expected 1`] = `
Object {
  "edges": Array [
    Object {
      "color": undefined,
      "from": 1,
      "label": undefined,
      "to": 0,
    },
    Object {
      "color": undefined,
      "from": 2,
      "label": undefined,
      "to": 0,
    },
    Object {
      "color": "rgba(255, 192, 84)",
      "from": 2,
      "label": undefined,
      "to": 1,
    },
    Object {
      "color": undefined,
      "from": 3,
      "label": undefined,
      "to": 1,
    },
    Object {
      "color": undefined,
      "from": 4,
      "label": undefined,
      "to": 1,
    },
    Object {
      "color": "rgba(255, 192, 84)",
      "from": 1,
      "label": undefined,
      "to": 2,
    },
    Object {
      "color": undefined,
      "from": 5,
      "label": undefined,
      "to": 2,
    },
    Object {
      "color": undefined,
      "from": 6,
      "label": undefined,
      "to": 2,
    },
    Object {
      "color": "rgba(255, 192, 84)",
      "from": 4,
      "label": undefined,
      "to": 3,
    },
    Object {
      "color": "rgba(255, 192, 84)",
      "from": 3,
      "label": undefined,
      "to": 4,
    },
    Object {
      "color": "rgba(255, 192, 84)",
      "from": 6,
      "label": undefined,
      "to": 5,
    },
    Object {
      "color": "rgba(255, 192, 84)",
      "from": 5,
      "label": undefined,
      "to": 6,
    },
    Object {
      "color": undefined,
      "from": 1,
      "label": undefined,
      "to": 7,
    },
    Object {
      "color": undefined,
      "from": 2,
      "label": undefined,
      "to": 7,
    },
  ],
  "nodes": Array [
    Object {
      "color": "rgba(133, 222, 255)",
      "id": 0,
      "label": "Harry Potter
1980 -",
      "level": 1,
    },
    Object {
      "color": "rgba(133, 222, 255)",
      "id": 1,
      "label": "James Potter
1960 - 1981",
      "level": 2,
    },
    Object {
      "color": "rgba(255, 189, 248)",
      "id": 2,
      "label": "Lily Potter
1960 - 1981",
      "level": 2,
    },
    Object {
      "color": "rgba(133, 222, 255)",
      "id": 3,
      "label": "Fleamont Potter",
      "level": 3,
    },
    Object {
      "color": "rgba(255, 189, 248)",
      "id": 4,
      "label": "Euphemia Potter",
      "level": 3,
    },
    Object {
      "color": "rgba(133, 222, 255)",
      "id": 5,
      "label": "Mr. Evans",
      "level": 3,
    },
    Object {
      "color": "rgba(255, 189, 248)",
      "id": 6,
      "label": "Mrs. Evans",
      "level": 3,
    },
    Object {
      "color": "rgba(255, 189, 248)",
      "id": 7,
      "label": "Gina Potter",
      "level": 1,
    },
  ],
}
`;
